src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public AbstractFigure getRandomFigure() {
+        Random random = new Random();
+        int figureType = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                double squareSide = random.nextDouble() * 10 + 1;
+                return new Square(color, squareSide);
```

Comment:
```suggestion
            case 0:
                return getRandomSquare();
```
lets make separate private methods to get each random figure, move `random.nextDouble() * 10 + 1` to a separate method as well